### PR TITLE
Add GithubRelease as package manager for MeshModel component generation

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: revolyssup <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: utils/github/error.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const ErrComponentGenerateCode = "11095"
+const ErrGetGHPackageCode = "11096"
+
+func ErrComponentGenerate(err error) error {
+	return errors.New(ErrComponentGenerateCode, errors.Alert, []string{"failed to generate components for the package"}, []string{err.Error()}, []string{}, []string{"Make sure that the package is compatible"})
+}
+
+func ErrGetGHPackage(err error) error {
+	return errors.New(ErrGetGHPackageCode, errors.Alert, []string{"Could not get the ArtifactHub package with the given name"}, []string{err.Error()}, []string{""}, []string{"make sure that the package exists"})
```
Comment (leecalcote) [inline utils/github/error.go:13]: No issues with GitHub rate limiting, right?

File: utils/github/package.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	"github.com/layer5io/meshkit/models"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/layer5io/meshkit/utils"
+	"github.com/layer5io/meshkit/utils/component"
+	"github.com/layer5io/meshkit/utils/manifests"
+)
+
+// This package is used when helm chart is fetched directly from github releases
+// the format https://github.com/<Owner>/<Repository>/releases/download/<Github-tag>/<Name-Version.tgz>
```
Comment (leecalcote) [inline utils/github/package.go:14]: Great comment.

File: utils/github/error.go (hunk 1 above)
Comment (Revolyssup) [reply utils/github/error.go:13]: No

**Review by leecalcote — APPROVED** [review]
LGTM

File: utils/github/package_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
```
Comment (leecalcote) [inline utils/github/package_test.go:7]: Uh-oh.. `io/util` has been deprecated

File: utils/github/package.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			return nil, fmt.Errorf("no versions found for given github package")
+		}
+		pkg.Version = versions[len(versions)-1]
+		fmt.Println("this: ", versions)
+	}
+	if pkg.FileName == "" {
+		pkg.FileName = fmt.Sprintf("%s.tgz", pkg.Version)
+	}
+	pkg.URL = fmt.Sprintf("https://github.com/%s/%s/releases/download/%s/%s", pkg.Owner, pkg.Repository, pkg.Version, pkg.FileName)
+	fmt.Println("url is", pkg.URL)
```
Comment (leecalcote) [inline utils/github/package.go:66]: We've not been using the MeshKit Logger, eh?

Comment (Revolyssup) [reply utils/github/package.go:66]: removing this redunant logs now.

File: utils/github/error.go [hunk 5]
```diff
@@ -10,5 +10,5 @@ func ErrComponentGenerate(err error) error {
 }
 
 func ErrGetGHPackage(err error) error {
-	return errors.New(ErrGetGHPackageCode, errors.Alert, []string{"Could not get the ArtifactHub package with the given name"}, []string{err.Error()}, []string{""}, []string{"make sure that the package exists"})
+	return errors.New(ErrGetGHPackageCode, errors.Alert, []string{"Could not get the Github package with the given name"}, []string{err.Error()}, []string{""}, []string{"make sure that the package exists"})
```
Comment (leecalcote) [inline utils/github/error.go:13]: What about GitHub not being reachable?

Comment (Revolyssup) [reply utils/github/error.go:13]: My laziness here. Okay making these error a little better and then I will  merge it. All lint and unit tests are passing

**Pull request merged**